packages/grid/test/drag-and-drop.common.js

```diff
@@ -1083,4 +1083,72 @@ describe('drag and drop', () => {
       expect(getFirstCell(grid).getAttribute('part')).to.contain('drag-source-row-cell');
     });
   });
+
+  describe('draggable grid', () => {
+    let container;
+    let items;
+
+    beforeEach(async () => {
+      container = fixtureSync(`
+      <div style="width: 400px; height: 400px;">
+        <vaadin-grid draggable="true" style="width: 300px; height: 300px;">
+          <vaadin-grid-column path="value"></vaadin-grid-column>
+        </vaadin-grid>
+      </div>
+    `);
+      grid = container.querySelector('vaadin-grid');
+      document.body.appendChild(container);
+      flushGrid(grid);
+      await nextFrame();
+      items = grid.shadowRoot.querySelector('#items');
+    });
+
+    async function setGridItems(count) {
+      grid.items = Array.from({ length: count }, (_, i) => ({ value: `Item ${i + 1}` }));
+      await nextFrame();
+    }
+
+    async function getMaxHeightDuringDragStart(element, items) {
+      let maxHeightDuringDragStart;
+      element.addEventListener(
+        'dragstart',
+        () => {
+          maxHeightDuringDragStart = items.style.maxHeight;
+        },
+        { once: true },
+      );
+      element.dispatchEvent(new DragEvent('dragstart'));
+      await new Promise((resolve) => {
+        requestAnimationFrame(resolve);
+      });
+      return maxHeightDuringDragStart;
+    }
+
+    it('should not modify maxHeight on dragstart for small grids', async () => {
+      await setGridItems(10);
+      const initialMaxHeight = items.style.maxHeight;
+      const maxHeightDuringDragStart = await getMaxHeightDuringDragStart(grid, items);
+      expect(maxHeightDuringDragStart).to.equal(initialMaxHeight);
+    });
+
+    ['5000', '50000'].forEach((count) => {
+      it('should temporarily set maxHeight to 0 on dragstart for large grids', async () => {
```

Comment:
These tests are checking the implementation detail of the fix (that max height gets set to the container) instead of producing the original issue (that the browser crashes on drag). If we ever change or remove the workaround in the future, these tests would fail for no good reason.

Test with something like this instead:
```js
it('should not crash when dragging a grid with lots of rows', async () => {
  const grid = fixtureSync(`
    <vaadin-grid draggable="true">
      <vaadin-grid-column path="name"></vaadin-grid-column>
    </vaadin-grid>
  `);
  grid.items = Array.from({ length: 5000 }, (_, i) => {
    return { name: `Item ${i}` };
  });

  await resetMouse();
  await hover(grid);
  await sendMouse({ type: 'down' });
  await sendMouse({ type: 'move', position: [100, 100] });
  await sendMouse({ type: 'up' });
});
```

Reply:
Updated the tests based on this suggestion. 

Note: The tests hang instead of fail/timeout without the fix. We decided to move forward with this approach.